### PR TITLE
Move to drone 0.5

### LINKED ISSUE #3890: Rebuild integration image according to best practices
As an engineer on VIC I want an integration image to be built using best practices in order to maximize the benefits of the image for CI testing.

https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/

### PR DESCRIPTION
**Pull request by jakedsouza**
Changes 

- Dockerfile.test changed to reduce number of layers and overall size 
- All drone yml files changed to work with drone 0.5 (except nightly) 
- local-integration-test.sh changed to work with new drone0.5 cli which does not accept the yaml from stdin, we now create a temp file and run the tests and then delete the temp file 
- Corresponding change in internal repo to add/update secrets to drone server (will push once this is merged) 

Fixes #3890 #4268 

New Issue - The build logs take a while to load in the ui especially towards the end of a test run. They load fast enough once a test is complete.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
-# to a specific .drone.yml so artifacts will not be uploaded to (OR DOWNLOADED
-# FROM) GCP in following builds without an updated .drone.sec
-#
-# When you are ready to submit a pull request, you must regenerate .drone.sec for the vmware/vic repo:
-#
-# $ drone -s https://ci.vmware.run -t ${DRONE_TOKEN} secure --in "secrets.yml" --out ".drone.sec" --repo vmware/vic --yaml ".drone.yml"
-#
+# After any change to this file you MUST re-sign and checkin the .drone.yml.sig
+# When you are ready to submit a pull request, you must regenerate .drone.yml.sig for the vmware/vic repo:
+# $ drone sign -s https://ci.vcna.io -t ${DRONE_TOKEN} vmware/vic
```
Comment (mhagen-vmware) [inline .drone.yml]: I thought drone 0.5 was supposed to manage this in a db now instead of our internal secrets yaml?

File: .drone.yml [hunk 2]
```diff
@@ -131,25 +130,22 @@ build:
       repo: vmware/vic
       event: [ push, tag ]
       branch: [ master, releases/*, refs/tags/* ]
-      success: true
+      status: success
 
-notify:
-  slack:
-    webhook_url: $$SLACK_URL
-    channel: mwilliamson-staff
```
Comment (mhagen-vmware) [inline .drone.yml]: is the channel not specified anywhere anymore?

File: .drone.yml [hunk 3]
```diff
@@ -131,25 +130,22 @@ build:
...
       branch: [ master, releases/*, refs/tags/* ]
-      failure: true
+      status: failure
 
-publish:
-  gcs:
-    auth_key: >
-      $$SERVICE_ACCOUNT_KEY
+  publish-gcs-builds:
+    image: maplain/drone-gcs:latest
```
Comment (mhagen-vmware) [inline .drone.yml]: where did this come from?

File: infra/integration-image/Dockerfile.test [hunk 4]
```diff
@@ -7,71 +7,78 @@
...
 RUN ln -s /usr/lib/vmware-ovftool/ovftool /usr/local/bin/ovftool
 
-RUN  wget http://mirrors.kernel.org/ubuntu/pool/main/w/wget/wget_1.18-2ubuntu1_amd64.deb
-RUN  dpkg -i wget_1.18-2ubuntu1_amd64.deb
+RUN curl -fsSLO https://releases.hashicorp.com/packer/0.12.2/packer_0.12.2_linux_amd64.zip && \
+    unzip packer_0.12.2_linux_amd64.zip -d /usr/bin && \
+	rm packer_0.12.2_linux_amd64.zip
 
-RUN apt-get install -y bzip2
+# RUN curl http://downloads.drone.io/drone-cli/drone_linux_amd64.tar.gz | tar zx
```
Comment (mhagen-vmware) [inline infra/integration-image/Dockerfile.test]: Keeping this for some reason?

File: .drone.yml (hunk 1 above)
Comment (jakedsouza) [inline .drone.yml]: Drone0.5 does keep secrets in its DB. 

You can not see the secrets though, just add,remove and list. We should still keep the secrets in our internal repo to version control any changes to secrets and also as a place to document all the secrets. 
I'll send out a separate change to the internal repo for the new secrets script once this is in.

File: .drone.yml (hunk 2 above)
Comment (jakedsouza) [inline .drone.yml]: The channel is not required. It is a part of the slack webhook https://vmware.slack.com/services/B59S91JFR

File: .drone.yml (hunk 3 above)
Comment (jakedsouza) [inline .drone.yml]: https://github.com/maplain/drone-gcs 

The current drone-gcs plugin does not work against 0.5, @maplain wrote a new plugin to work with 0.5

File: infra/integration-image/Dockerfile.test (hunk 4 above)
Comment (jakedsouza) [inline infra/integration-image/Dockerfile.test]: Will remove

File: infra/integration-image/Dockerfile.test [hunk 5]
```diff
@@ -7,71 +7,77 @@
...
-RUN apt-get install sshpass
-
-RUN apt-get install -yq ant openjdk-7-jdk
-RUN apt-get install -yq rpcbind nfs-common
-RUN apt-get install unzip
-RUN apt-get install zip
-
-RUN curl -L https://github.com/docker/compose/releases/download/1.11.2/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
-RUN chmod +x /usr/local/bin/docker-compose
+RUN curl -sL https://deb.nodesource.com/setup_7.x | bash - && \
```
Comment (anchal-agrawal) [inline infra/integration-image/Dockerfile.test]: Thanks for doing this 👍

**Pull request merged**